packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -64,15 +93,105 @@ export default class Sayt extends LitElement {
     this.visible = true;
   }
 
+  /**
+   * Makes SAYT visible if the event refers to the correct SAYT component.
+   * @see [[isCorrectSayt]]
+   *
+   * @param event An event that can contain a searchbox ID.
+   */
+  showCorrectSayt(event: CustomEvent) {
+    if (this.isCorrectSayt(event)) {
+      this.showSayt();
+    }
+  }
+
   /**
    * Changes the `visible` property to `false`.
    */
   hideSayt() {
     this.visible = false;
   }
 
+  /**
+   * Hides SAYT if the event refers to the correct SAYT component.
+   *
+   * @param event An event that can contain a searchbox ID.
+   */
+  hideCorrectSayt(event: CustomEvent) {
+    if (this.isCorrectSayt(event)) {
+      this.hideSayt();
+    }
+  }
+
+  /**
+   * Determines whether an event refers to the correct SAYT. This is true if
+   * a matching `searchbox` ID is specified, if the event has no `searchbox`
+   * ID specified, or if SAYT has no `searchbox` ID specified.
+   *
+   * @param event An event that can contain a searchbox ID for comparison.
+   */
+  isCorrectSayt(event: CustomEvent): boolean {
+    const searchbox = event.detail && event.detail.searchbox;
+    return !searchbox || !this.searchbox || searchbox === this.searchbox;
+  }
+
+  /**
+   * Processes a click event in order to close SAYT under the right conditions.
+   * 
+   * @param event The click event.
+   */
+  processClick(event: MouseEvent) {
+    const target = event.target as Node;
+    if (this.contains(target) || this.nodeInSearchbox(target)) return;
```

Comment:
If I return something here (eg. `false`), I'd have to `return true` at the end. Given that its job is to process a click event, I personally don't think there is much point in returning anything. I'm only returning to prevent future logic from running.

Reply:
> Perhaps a boolean to indicate that the click as been processed in an expected manner?

If it had been processed in an unexpected manner, it would be better to throw an exception instead of relying on a return code.